### DISCUSSION TITLE
How to set up the Miele integration

### DISCUSSION THREAD

**Discussion by astrandb** [astrandb/miele#366] [Q&A]
## Installation
Download the Miele integration with HACS or any other method you prefer. Restart Home Assistant

## Here are a number of screen-shots describing a successful installation:

![MieleInstall1](https://github.com/astrandb/miele/assets/5016010/1fb8f818-670f-41b2-bc7f-ec3aa8d82f8d)

Hit Configure or if you dont see this panel you can hit Add integration and select "Miele"

![MieleInstall2](https://github.com/astrandb/miele/assets/5016010/69f78ff2-3aea-4af3-b860-b84f5de318b8)

Submit to continue

![MieleInstall3](https://github.com/astrandb/miele/assets/5016010/b84ddb38-a6a0-46ff-9a38-01e576c0876a)

Fill in your credentials and press Add

![MieleInstall4](https://github.com/astrandb/miele/assets/5016010/8fe2e90c-81fc-4406-84be-7e75280d4222)

You should be directed to login page at Miele. Enter your login details. The username/password are the same as you use in the Miele app. Note that there has been several reports of failed logins when the password contains special characters even though it works in other areas with Miele.

![MieleInstall5](https://github.com/astrandb/miele/assets/5016010/f2fac53e-cdfc-4df7-a5f1-eacd697a7ba7)

Make sure that the suggested instance URL is correct in your specific context. If not it must be corrected before continuing. This typically happens if you have multiple HA instances or if the there is a non-standard network setup.

![MieleInstall6](https://github.com/astrandb/miele/assets/5016010/24c00cae-afdd-4152-bf13-bc8cfa948eae)

Optionally select area and press Finish.

![MieleInstall7](https://github.com/astrandb/miele/assets/5016010/b8664467-435c-4289-a90a-d4d30a6e8ce2)

If you suspect that you have entered wrong client_id or wrong client_secret you can remove the credentials by going to Settings-Integrations and selecting the overflow menu (tree dots in upper right corner).

**Comment by picardi21** [comment]
to me after being redirected, I enter the credentials, it goes back to home assistant and tells me that I am correctly logged in, but it does not show me any entity

**Reply by davsoliman** [reply]
Im having a similar issue. It says "Re-authentication was successful" and then nothing. Is this what's happening to you?

**Reply by Hubi-Swiss** [reply]
Same issue here. I can't see the integration or any device. I found only an Entity for update: See attachment
<img width="1231" alt="Screenshot 2024-09-28 at 11 29 50" src="https://github.com/user-attachments/assets/61b24b1e-6a3a-4b98-bdbb-d308b4656576">

**Comment by JNSwanson** [comment]
When I try to configure I get:
Error
Config flow could not be loaded: {"message":"Invalid flow specified"}

**Comment by CNjord** [comment]
Im have  problem when install this integration. After filling in my credentials I dont get directed to Mieles login page, Nothing happens.
Ive tried sevral times deleting the integration and autentisering id then reboot and try with new credentials same result.t

**Comment by jasonhollis** [comment]
I'm helping someone with this integration.  I've noted that there are reports of the authentication page having problems with special characters:

![image](https://github.com/user-attachments/assets/8508ac4f-0aa3-4a6a-b819-08d391df3c2d)

If you try to change the password in the app to something without special characters the app won't let you.  It requires at least one special character.

**Reply by astrandb** [reply]
Well, the problem has been with accented charactetrs like "åäö" and not with "#$&".
